### PR TITLE
Unit test for statHttp within Pelican Client

### PR DESCRIPTION
**Pull request by joereuss12**
Added unit test for statHttp within Pelican client. Note: this should not be merged until #884 is completed since currently statHttp only works with topology. Therefore, I expect tests to fail and I will readdress them once that PR is merged and this function is working properly.

### CODE REVIEW

**Comment by joereuss12** [comment]
Note: the osdf:// stat test is currently a pelican url. This is waiting for #878 to be merged before fixing. Made issue #961 to do this

File: client/fed_test.go [hunk 1]
```diff
@@ -407,3 +407,87 @@ func TestGetPublicRead(t *testing.T) {
...
+	common.ResetOriginExports()
+
+	fed := fed_test_utils.NewFedTest(t, bothPublicOriginCfg)
+
+	t.Run("testStatHttpPelicanScheme", func(t *testing.T) {
+		testFileContent := "test file content"
+		// Drop the testFileContent into the origin directory
+		tempFile, err := os.Create(filepath.Join(((*fed.Exports)[0]).StoragePrefix, "test.txt"))
+		assert.NoError(t, err, "Error creating temp file")
+		defer os.Remove(tempFile.Name())
```
Comment (haoming29) [inline client/fed_test.go:424]: I saw that `NewFedTest` removes all files at the end of the test, so this is not needed.

File: client/fed_test.go [hunk 2]
```diff
@@ -407,3 +407,87 @@ func TestGetPublicRead(t *testing.T) {
...
+			assert.Equal(t, int64(17), int64(objectSize))
+		}
+	})
+
+	t.Run("testStatHttpOSDFScheme", func(t *testing.T) {
+		testFileContent := "test file content"
+		// Drop the testFileContent into the origin directory
+		tempFile, err := os.Create(filepath.Join(((*fed.Exports)[0]).StoragePrefix, "test.txt"))
+		assert.NoError(t, err, "Error creating temp file")
+		defer os.Remove(tempFile.Name())
```
Comment (haoming29) [inline client/fed_test.go:449]: same here

File: client/fed_test.go [hunk 3]
```diff
@@ -407,3 +407,87 @@ func TestGetPublicRead(t *testing.T) {
...
+			assert.Equal(t, int64(17), int64(objectSize))
+		}
+	})
+
+	t.Run("testStatHttpIncorrectScheme", func(t *testing.T) {
+		testFileContent := "test file content"
+		// Drop the testFileContent into the origin directory
+		tempFile, err := os.Create(filepath.Join(((*fed.Exports)[0]).StoragePrefix, "test.txt"))
+		assert.NoError(t, err, "Error creating temp file")
+		defer os.Remove(tempFile.Name())
```
Comment (haoming29) [inline client/fed_test.go:475]: here

**Review by haoming29 — CHANGES_REQUESTED** [review]
LGTM with one small change requested

**Review by haoming29 — APPROVED** [review]
LGTM!

**Pull request merged**